### DISCUSSION TITLE
Uploading file from Web Browser on Android Phone

### DISCUSSION THREAD

**Discussion by ozturkahmetcevdet** [Hieromon/AutoConnect#489] [General]
Hello Folks,
First of all, thanks for this beautiful library. The interface is great and very easy to use and stable.

"Choose file" (AutoConnectUploadHandler -> HTTPUpload) does not bring up the file selection page when I want to "update" the web page via Portal on Android phone. But it works fine on other operating systems and computers.

Is there a way you can suggest me?

**Comment by Hieromon** [comment]
@ozturkahmetcevdet What is "Choose file" (AutoConnectUploadHandler -> HTTPUpload), is it regular file upload by [AutoConnectOTA](https://hieromon.github.io/AutoConnect/otabrowser.html#regular-file-uploading-using-autoconnectotaenhanced-wv120)? Your question is too vague and I don't know how to answer it.

If your sketch complies with the AutoConnect specification, but you have a question about its behavior, please provide specifics. In that case, please follow the submission form you are guided through in the [documentation](https://hieromon.github.io/AutoConnect/otabrowser.html#regular-file-uploading-using-autoconnectotaenhanced-wv120). It will help us resolve the issue quickly.

Incidentally, the HTML implementation of the "Choose file" button for AutoConnectOTA (which appears as an item in the AutoConnect menu via the **AutoConnectConfig::ota** setting) is the `input type="file"` element. You can check the support status of this HTML element for each browser at the following link, regardless of OS or device:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#browser_compatibility

**EDIT**
If your reference is to the Chromium-based WebView, then it is a known issue and has nothing to do with AutoConnect. It is an external issue in the Google Chromium web view. More information about this is available here:

- [File upload broken as a Chromium bug (Chromium issue tracker)](https://bugs.chromium.org/p/chromium/issues/detail?id=278640)
- [File input element not opening file picker in Android 4.4 (Apache Cordova issue tracker)](https://issues.apache.org/jira/browse/CB-5294)
- [HTML file input in Android webview (4.4 KitKat) (Stack Overflow)](https://stackoverflow.com/questions/19882331/html-file-input-in-android-webview-android-4-4-kitkat)

And for now, I do not intend to employ any element other than input type=file for AutoConnectOTA upload file selection.

**Comment by Hieromon** [comment]
The adoption of input type=file in AutoConnectOTA and the non-support of this HTML element in Android WebView is a specification conflict that should be brought to the attention of other users. Therefore, this topic is moved to Discussions.

**Comment by ozturkahmetcevdet** [comment]
Thank you for your support.
